cmake_mtime_md5.py

```diff
@@ -0,0 +1,72 @@
+import time
+import os
+from hashlib import md5
+
+# #
+# Functions
+# #
+
+# get the md5 hash of a file's contents
+def getMD5Hash(file):
+    src_file = open(file, "r", encoding="utf-8")
+    file_contents = src_file.read()
+    src_file.close()
+
+    md5hash = md5()
+    md5hash.update(file_contents.encode("utf8"))
+    return md5hash.hexdigest()
+
+# recursively get all files in all subdirectories
+def getAllFiles(start, file_list):
+    for root, directories, files in os.walk(start):
+        for directory in directories:
+            getAllFiles(start+"/"+directory, file_list)
+        
+        for file in files:
+            file_list.append(root+"/"+file)
+
+# #
+# Variables
+# #
+today = time.time()
+yesterday = time.time() - (24*60*60)
+mtime_file = "build/cmake_mtime_md5.txt"
+src_list = []
+
+# #
+# Run
+# #
+getAllFiles("src", src_list)
+
+# get the current md5 hash of all relevant files in src/, and set their mtime to yesterday
+hash_dict_curr = {}
+for file in src_list:
+    os.utime(file, (yesterday, yesterday))
+    if (file.endswith(".cpp") or file.endswith(".hpp") or file.endswith(".tpp") or file.endswith(".c") or file.endswith(".h")) and "CMakeFiles" not in file:
+        hash_dict_curr.update({file : getMD5Hash(file)})
+
+# if there's an existing stored list of md5 hashes, read that in
+if os.path.isfile(mtime_file):
+    readfile = open(mtime_file, "r", encoding="utf-8")
+    hash_dict_prev = {}
+    
+    for line in readfile:
+        line_list = line.strip().split(":")
+        hash_dict_prev.update({line_list[0] : line_list[1]})
+    
+# compare file hashes between the set of hashes from the previous run and the current run
+# if the file has changed, or is brand new, set its mtime to today so that cmake will know to recompile it
+    for key in hash_dict_curr.keys():
```

Comment:
Because you use both keys and values, I'd prefer if you used `hash_dict_curr.items()`.